eppo_core/src/events/event_delivery.rs

```diff
@@ -0,0 +1,60 @@
+use crate::events::event::Event;
+use crate::Error;
+use log::info;
+use reqwest::{Method, StatusCode};
+use serde::{Deserialize, Serialize};
+
+pub struct EventDelivery {
+    sdk_key: String,
+    ingestion_url: String,
+    client: reqwest::Client,
+}
+
+#[derive(serde::Deserialize)]
+struct EventDeliveryResponse {
+    failed_events: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct IngestionRequestBody {
+    eppo_events: Vec<Event>,
+}
+
+/// Responsible for delivering event batches to the Eppo ingestion service.
+impl EventDelivery {
+    pub fn new(sdk_key: String, ingestion_url: String) -> Self {
+        let client = reqwest::Client::new();
+        EventDelivery {
+            sdk_key,
+            ingestion_url,
+            client,
+        }
+    }
+
+    // Delivers the provided event batch and returns a Vec with the events that failed to be delivered.
+    pub async fn deliver(self, events: &[Event]) -> Result<Vec<String>, Error> {
```

Comment:
minor: `EventDeliveryResponse` is more specific than a generic `Vec<String>`. (When I see `Vec<String>`, I have no idea what it is.)
```suggestion
    pub async fn deliver(self, events: &[Event]) -> Result<EventDeliveryResponse, Error> {
```